lib.go

```diff
@@ -80,21 +80,21 @@ func NewTrieNode(data *C.char, sz C.int) C.uintptr_t {
 	}
 
 	// calculate key for caching
-	if _, err := n.Key(); err != nil {
+	if _, err := n.NodeHash(); err != nil {
 		return 0
 	}
 
 	return C.uintptr_t(cgo.NewHandle(n))
 }
 
 // obtain the key hash, must be free by caller
-//export TrieNodeKey
-func TrieNodeKey(pN C.uintptr_t) unsafe.Pointer {
+//export TrieNodeHash
+func TrieNodeHash(pN C.uintptr_t) unsafe.Pointer {
 	h := cgo.Handle(pN)
 	n := h.Value().(*trie.Node)
 
-	k, _ := n.Key()
-	return C.CBytes(k.Bytes())
+	hash, _ := n.NodeHash()
```

Comment:
Maybe not so critical since the method is mainly used in testing.

Also the error would raise under very rare case, and it should still be catched by the caller for returning nil pointer.